app/src/main/java/edu/wisc/ece/pinpoint/pages/newpin/NewPinImageContentFragment.java

```diff
@@ -1,18 +1,145 @@
+import android.content.Intent;
+import android.net.Uri;
 import android.os.Bundle;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.Looper;
+import android.provider.MediaStore;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-
+import android.widget.ImageView;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
 import androidx.fragment.app.Fragment;
-
+import com.bumptech.glide.Glide;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import edu.wisc.ece.pinpoint.BuildConfig;
 import edu.wisc.ece.pinpoint.R;
 
 public class NewPinImageContentFragment extends Fragment {
+
+    private int imageOpenType = 0;
+    private ActivityResultLauncher<Intent> photoTakerLauncher;
+    private ActivityResultLauncher<Intent> photoPickerLauncher;
+    private File photoFile;
+    private Handler h;
+    private Runnable takePictureRunnable;
+    private Runnable uploadPictureRunnable;
+    public Uri photo_uri;
+    private ImageView imageContentUpload;
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         return inflater.inflate(R.layout.fragment_new_pin_image_content, container, false);
     }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        photoPickerLauncher =
+                registerForActivityResult(new ActivityResultContracts.StartActivityForResult(),
+                        result -> {
+                            if (result.getResultCode() == RESULT_OK && result.getData() != null) {
+                                photo_uri = result.getData().getData();
+                                Glide.with(requireContext()).load(photo_uri).centerCrop()
+                                        .into(imageContentUpload);
+                            }
+                        });
+
+        photoTakerLauncher =
+                registerForActivityResult(new ActivityResultContracts.StartActivityForResult(),
+                        result -> {
+                            if (result.getResultCode() == RESULT_OK && result.getData() != null) {
+                                Glide.with(this).load(photo_uri).centerCrop()
+                                        .into(imageContentUpload);
+                            }
+                        });
+    }
+
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        h = new Handler(Looper.getMainLooper());
+        takePictureRunnable = () -> takePicture(getView());
```

Comment:
these could be just 
```java
takePictureRunnable = this::takePicture;
uploadPictureRunnable = this::uploadPicture;
```
since `takePicture` and `uploadPicture` don't actually use the `view` parameter.